ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -274,7 +289,22 @@
     executorService.submit(() -> {
       // TODO introduce failover handling (retry on remote datacenters, handle SQL exceptions)
       try (Connection connection = dcToDataSource.get(localDatacenter).getConnection()) {
-        future.complete(transaction.run(account.getId(), container.getId(), connection));
+        T result;
+        if (autoCommit) {
+          result = transaction.run(account.getId(), container.getId(), connection);
+        } else {
+          connection.setAutoCommit(false);
+          try {
+            result = transaction.run(account.getId(), container.getId(), connection);
+            connection.commit();
```

Comment:
Can you give me a brief introduction about what's the connection.commit() and connection.rollback() are doing? a little bit confuse about it. Thank you!

Reply:
I will update this. The way I am using variables is a little wonky and soon to be deprecated behavior in mysql. I think I will update the delete query to do a 2 step transaction:
1. select the current row to give us the blob ID
2. if the current row is present and not already deleted, update (set deletion time)
Since this is two operations that we want to be an atomic unit we have to manually commit the transaction like so.